src/Aspire.Hosting.Azure.CosmosDB/api/Aspire.Hosting.Azure.CosmosDB.cs

```diff
@@ -37,12 +57,38 @@ public static partial class AzureCosmosExtensions
 
 namespace Aspire.Hosting.Azure
 {
+    public partial class AzureCosmosDBContainerResource : ApplicationModel.Resource, ApplicationModel.IResourceWithParent<AzureCosmosDBDatabaseResource>, ApplicationModel.IResourceWithParent, ApplicationModel.IResource, ApplicationModel.IResourceWithConnectionString, ApplicationModel.IManifestExpressionProvider, ApplicationModel.IValueProvider, ApplicationModel.IValueWithReferences, IResourceWithAzureFunctionsConfig
+    {
+        public AzureCosmosDBContainerResource(string name, string containerName, string partitionKeyPath, AzureCosmosDBDatabaseResource parent) : base(default!) { }
+
+        public ApplicationModel.ReferenceExpression ConnectionStringExpression { get { throw null; } }
+
+        public string ContainerName { get { throw null; } set { } }
+
+        public AzureCosmosDBDatabaseResource Parent { get { throw null; } }
+
+        public string PartitionKeyPath { get { throw null; } set { } }
+
+        void IResourceWithAzureFunctionsConfig.ApplyAzureFunctionsConfiguration(System.Collections.Generic.IDictionary<string, object> target, string connectionName) { }
+    }
+
+    public partial class AzureCosmosDBDatabaseResource : ApplicationModel.Resource, ApplicationModel.IResourceWithParent<AzureCosmosDBResource>, ApplicationModel.IResourceWithParent, ApplicationModel.IResource, ApplicationModel.IResourceWithConnectionString, ApplicationModel.IManifestExpressionProvider, ApplicationModel.IValueProvider, ApplicationModel.IValueWithReferences, IResourceWithAzureFunctionsConfig
+    {
+        public AzureCosmosDBDatabaseResource(string name, string databaseName, AzureCosmosDBResource parent) : base(default!) { }
+
+        public ApplicationModel.ReferenceExpression ConnectionStringExpression { get { throw null; } }
+
+        public string DatabaseName { get { throw null; } set { } }
+
+        public AzureCosmosDBResource Parent { get { throw null; } }
+
+        void IResourceWithAzureFunctionsConfig.ApplyAzureFunctionsConfiguration(System.Collections.Generic.IDictionary<string, object> target, string connectionName) { }
+    }
+
     public partial class AzureCosmosDBEmulatorResource : ApplicationModel.ContainerResource, ApplicationModel.IResource
     {
         public AzureCosmosDBEmulatorResource(AzureCosmosDBResource innerResource) : base(default!, default) { }
 
         public override ApplicationModel.ResourceAnnotationCollection Annotations { get { throw null; } }
-
-        public override string Name { get { throw null; } }
```

Comment:
@eerhardt is this a breaking change?

Reply:
It shouldn't be.

https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules#members

> ✔️ ALLOWED: Adding or removing an override

> Introducing an override might cause previous consumers to skip over the override when calling [base](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/base).